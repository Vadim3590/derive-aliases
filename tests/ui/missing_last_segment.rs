pub mod derive_alias {
    derive_aliases::define! {
        Eq = ::core::cmp::Eq, ::core::cmp::;
        //                                  ^ missing last segment
        AndMore = ::std::hash::Hash, ..Ord, ::core::clone::Clone, ::core::marker::Copy, ::core::default::Default;
    }
}

fn main() {}
