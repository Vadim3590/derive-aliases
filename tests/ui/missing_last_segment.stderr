error: expected identifier to form a path like `::std::hash::Hash`
 --> tests/ui/missing_last_segment.rs:3:44
  |
3 |         Eq = ::core::cmp::Eq, ::core::cmp::;
  |                                            ^

error[E0432]: unresolved import `crate`
 --> tests/ui/missing_last_segment.rs:2:5
  |
2 | /     derive_aliases::define! {
3 | |         Eq = ::core::cmp::Eq, ::core::cmp::;
4 | |         //                                  ^ missing last segment
5 | |         AndMore = ::std::hash::Hash, ..Ord, ::core::clone::Clone, ::core::marker::Copy, ::core::default::Default;
6 | |     }
  | |_____^ no `AndMore` in `derive_alias`
  |
  = note: this error originates in the macro `derive_aliases::define` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `Ord` in `derive_alias`
 --> tests/ui/missing_last_segment.rs:5:40
  |
5 |         AndMore = ::std::hash::Hash, ..Ord, ::core::clone::Clone, ::core::marker::Copy, ::core::default::Default;
  |                                        ^^^ could not find `Ord` in `derive_alias`
